src/m365/spo/commands/file/file-get.ts

```diff
@@ -84,35 +84,53 @@ class SpoFileGetCommand extends SpoCommand {
         'accept': 'application/json;odata=nometadata'
       },
       // Set responseType to arraybuffer, otherwise binary data will be encoded
-      // to utf8 and binary data is corrupt 
+      // to utf8 and binary data is corrupt
       responseType: args.options.asFile ? 'arraybuffer' : 'json'
     };
 
-    request
-      .get<string>(requestOptions)
-      .then((file: string): void => {
-        if (args.options.asString) {
-          logger.log(file.toString());
-        }
-        else if (args.options.asListItem) {
-          const fileProperties: FileProperties = JSON.parse(JSON.stringify(file));
-          logger.log(fileProperties.ListItemAllFields)
-        }
-        else if (args.options.asFile) {
-          if (args.options.path) {
-            fs.writeFileSync(args.options.path, file);
-            if (this.verbose) {
-              logger.log(`File saved to path ${args.options.path}`);
-            }
+    if (args.options.asFile && args.options.path) {
+      request
+        .get<any>(requestOptions, true)
+        .then((file: any): Promise<string> => {
+          return new Promise((resolve, reject) => {
+            const writer = fs.createWriteStream(args.options.path as string);
+
+            file.data.pipe(writer);
```

Comment:
Is this correct? In the `request`, we resolve the Promise with raw data without wrapping it in a `data` property. Shouldn't we then here be calling `file.pipe` or is `.data` coming actually from Axios?

Reply:
Based on https://github.com/axios/axios#example and snippet: 
``` 
// GET request for remote image in node.js
axios({
  method: 'get',
  url: 'http://bit.ly/2mTM3nY',
  responseType: 'stream'
})
  .then(function (response) {
    response.data.pipe(fs.createWriteStream('ada_lovelace.jpg'))
  });
```

I used the the .data.pipe

Reply:
Cool, good to double check 👍